docs/overview.rst

```diff
@@ -0,0 +1,56 @@
+--------------------
+The program intent and engagement (PIE) service is a data store used to capture
+various interactions a user could have with a program and its courses.
+
+Users of the service can query it for a user's intents, or create new intents
+via HTTP request or management command.
+
+Program Intent Model
+~~~~~~~~~~~~~~~~~~~~
+The program intent model is meant to represent when a learner marked an interest in a program, why we believe
+their action represents an intent, and how certain we are of their intent to participate in a program.
+
+The types of intent reasons and their corresponding certainty are described in the following table
+
++----------------------+--------------------------------------------+-------------+
+| Reason               | Description                                | Certainty   |
++======================+============================================+=============+
+| BUNDLE_PURCHASE      | User purchased program as a bundle         | CERTAIN_YES |
++----------------------+--------------------------------------------+-------------+
+| ENROLLED_ALL_COURSES | User has enrolled in all program courses   | MAYBE       |
++----------------------+--------------------------------------------+-------------+
+| EARNED_PROGRAM_CERT  | User has earned a program certificate      | CERTAIN_YES |
++----------------------+--------------------------------------------+-------------+
+
+Accessing Learner Intents
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Learner intents can be queried via HTTP GET request. The request user will be used to filter
+intents, so only intents associated with that user will be returned.
+
+Currently the implementation will only return the most recent and most certain intent
+per program for each user.
+
+Inserting Learner Intents
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are two methods to create new learner intents:
+
+- HTTP POST request
+- Insert management command
+
+A learner intent can be created via HTTP POST request. The API expects the request data to contain::
+
+    program_uuid: program UUID to be associated with the attempt
+    reason: reason we believe the learner intends to participate in the program
+    certainty: how certain we are that the reason corresponds to a learner's intentions
+    effective_timestamp: datetime for when the intent was expressed (e.g. datetime of when a learner purchased a program as a bundle)
+
+If a user attempts to create a duplicate intent via HTTP request, a 400 response will be returned.
+
+Learner intents can also be bulk created via `bulk_insert_intents` management command. The command expects a CSV with a
+header row as input, with each row defined in the CSV representing an intent that should be created. The expected columns for
+the data are::
+
+    program_uuid, user_id, intent_reason, intent_certainty, effective_time
+
+If a duplicate intent is defined in the CSV file, the management command will not attempt to create a new intent for that duplicate data.
```

Comment:
Hey @alangsto! @ashultz0 and I were discussing this in the insert endpoint pull request: he'd said that because both of the workarounds to suppress the 400 error for duplicates could be buggy, that we should stick with having the 400 error. This is probably something that should be documented here. Lemme know if you have any questions/thoughts!

Reply:
I have that documented for intents created via HTTP POST request - the behavior of the management is slightly different in that it essentially "swallows" duplicates

Reply:
Oops, nvm! Just seeing you have this already :)